### PR TITLE
feat: [sc-156683] [SAP-OData] Add search query field

### PR DESCRIPTION
**Pull request by alexbourret**
Story details: https://app.shortcut.com/dataiku/story/156683

### CODE REVIEW

File: python-connectors/sap_odata/connector.json [hunk 1]
```diff
@@ -64,9 +64,18 @@
             "visibilityCondition": "model.show_advanced_parameters == true"
         },
         {
-            "name": "bulk_size",
-            "label": "Bulk size",
+            "name": "odata_filter_query",
+            "label": "Filter query",
             "description": "",
```
Comment (Alexlandeau) [inline python-connectors/sap_odata/connector.json:69]: 🧑‍🍳 suggestion: as I struggled a bit to find the filter syntax, I would add a description to specify that it has to match the sap syntax (and not the DSS one), maybe with a link to https://help.sap.com/docs/SAP_IoT/fffd6ca18e374c2e80688dab5c31527f/075e41f5538c4a6fa1176db1ba03f602.html, or with a mention in the plugin's doc

**Review by Alexlandeau — COMMENTED** [review]
Code is fine and filters work as intended. However, would improve a bit documentation, what do you think ?

File: python-connectors/sap_odata/connector.json (hunk 1 above)
Comment (alexbourret) [reply python-connectors/sap_odata/connector.json:69]: 👍  Added the link in [3cec921](https://github.com/dataiku/dss-plugin-sap-odata/pull/7/commits/3cec9215ce905e55fa80b89dca246a043bcddcb8)

**Review by Alexlandeau — APPROVED** [review]
LGTM !

**Pull request merged**